server/routes/probationPractitionerReferrals/interventionProgressPresenter.ts

```diff
@@ -39,6 +39,20 @@ export default class InterventionProgressPresenter {
     return this.referral.assignedTo !== null
   }
 
+  get referralEndRequested(): boolean {
+    return this.referral.endRequestedAt !== null
+  }
+
+  get referralEndRequestedText(): string {
```

Comment:
So I wasn't sure if the logic for constructing the string should be in the Presenter or the View class. 

My understanding of the roles are:
- Presenter: derive fields necessary for constructing a page
- View: format fields into the form that the page expects that fields to be

So in my PR I just created a new method which passes the endedFields and the View was responsible for formatting that into a form that is required for the cancellation banner.

I suppose the benefit of having it in the Presenter class is that it is testable since there are no unit tests for the View classes.

Reply:
We've been a bit inconsistent in the codebase. The way I see it is - presenter does most of the heavy lifting/logic for wrangling data fields into text if we need to interpolate them somewhere, and also contains some text to be passed to the view.

I guess we have a similar understanding, but possibly different interpretations of formatting fields - I guess I'd see formatting fields with HTML as a View concern, but for shorter strings without any formatting I'd opt for the presenter.

I'm not 100% on this, it'd be good to figure out a consistent way forward!

Reply:
I see. I think so long as we're consistent is all that matters. 
Given I'm completely new I'll make sure to follow your conventions - I'll make the changes to my last PR to ensure that the formatting is done in the presenter class.